### PR TITLE
one level frustum culling

### PR DESCRIPTION
**Pull request by barlebino**

### CODE REVIEW

File: src/Engine/Scene/Scene.hpp [hunk 1]
```diff
@@ -79,24 +77,18 @@ class Scene {
...
-template <typename CompT, typename... Args>
-CompT & Scene::createComponent(Args &&... args) {
-    return addComponent(std::unique_ptr<CompT>(new CompT(std::forward<Args>(args)...)));
-}
-
 template <typename CompT>
 CompT & Scene::addComponent(std::unique_ptr<CompT> component) {
     CompT & comp(*component);
-    m_componentInitQueue[std::type_index(typeid(CompT::SystemClass))].emplace_back(std::move(component));
+    m_componentInitQueue[std::type_index(typeid(typename CompT::SystemClass))].emplace_back(std::move(component));
```
Comment (Daskie) [inline src/Engine/Scene/Scene.hpp]: why did you add typename here?

Comment (barlebino) [reply src/Engine/Scene/Scene.hpp]: Don't know, this is the error I get when it is not there

error: dependent-name ‘CompT:: SystemClass’ is parsed as a non-type, but instantiation yields a type

Comment (Daskie) [reply src/Engine/Scene/Scene.hpp]: huh, what compiler are you using?

Comment (Daskie) [reply src/Engine/Scene/Scene.hpp]: i made that change across the board, hopefully shouldn't have any other issues

Comment (barlebino) [reply src/Engine/Scene/Scene.hpp]: thanks. g++

File: src/Engine/System/RenderSystem.hpp [hunk 2]
```diff
@@ -15,7 +15,9 @@
 #include "Shaders/DiffuseShader.hpp"
 #include "Shaders/BounderShader.hpp"
 
-
+#include <map>
+#include <string>
+#include <iostream>
```
Comment (jaafersheriff) [inline src/Engine/System/RenderSystem.hpp]: Please remove redundant #includes. 
string is included twice and iostream shouldn't need to be included in headers.

Comment (barlebino) [reply src/Engine/System/RenderSystem.hpp]: We can remove map and string, but removing iostream gives me error: ‘cerr’ is not a member of ‘std’

Comment (jaafersheriff) [reply src/Engine/System/RenderSystem.hpp]: You right, iostream needs to stay included. I think same with map.
string is included twice tho
